fix(autofill-colours): remove yellow autofill box and styling fixes

jor-row opened the pull request:
# Description

This PR styles the autofill icon in Safari so that it is visible in our dark theming - the same changes as made in https://sqbu-github.cisco.com/WebExSquared/webex-web-client/pull/3032 but generalising them to the momentum component. It also removes the yellow autofill box on Safari (except in the case of a suggested password) and Firefox, making it transparent. In the case of a suggested password in Safari the yellow box is restyled to have a border radius which aligns with that of the TextInput.

# Links
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-362673

Safari ->
Before
![SAFARI-BEFORE](https://user-images.githubusercontent.com/86778628/191272800-b835367b-ad39-4472-a2e0-e5ba2368b4e5.png)
After 
![SAFARI_AFTER](https://user-images.githubusercontent.com/86778628/191272837-7e709fd3-699a-4bac-8002-d527e05c2ce9.png)

Firefox -> 
Before
<img width="534" alt="FIREFOX-BEFORE" src="https://user-images.githubusercontent.com/86778628/191272902-f4406eb2-d1b8-4e65-bd5e-e57e9b9dc032.png">
After
<img width="549" alt="FIREFOX-AFTER" src="https://user-images.githubusercontent.com/86778628/191272928-59edebfd-a995-47d2-958f-addbc6158743.png">

maxinteger reviewed (changes requested):
Could you please create a new story in storybook with a login form to test these scenarios?

for example 
```
const LoginForm = () => (
  <form>
    <PaddedExample label={'Email'} type={'email'} />
    <PaddedExample label={'Password'} type={'password'} />
    <button type={'submit'}>Submit</button>
  </form>
);
```

Probably you can add some text as well to explain why it is needed and you can use the Momentum button for submit :)

Also the fix does not work when you ask Safari to suggest a new password:
![image](https://user-images.githubusercontent.com/983050/191282128-d6e5b52e-dab1-44af-a1a3-e289285a8cd2.png)

jor-row commented:
@maxinteger I've added in two 'form' stories to storybook. This is the first time I've added stories to a component so any improvements or things I've missed are more than welcome!

maxinteger commented on src/components/TextInput/TextInput.stories.tsx:
```diff
@@ -85,10 +92,64 @@ const BetterExample: FC<StoryProps> = (props: StoryProps) => {
+
+  return (
+    <>
+      <form>
```
Please add the following:
```suggestion
    <form
      onSubmit={(e) => {
        e.preventDefault();
        location.reload();
      }}
    >
```
this will prevent to reload the whole storyboard, but it will reload the `iframe` of the story which triggers the browser password manager to store the email / password

maxinteger commented on src/components/TextInput/TextInput.style.scss:
```diff
@@ -39,6 +39,48 @@
       color: var(--textinput-placeholder-text);
     }
+
+    &::-webkit-contacts-auto-fill-button {
```
I just noticed, the background of the input changes after autofill in chrome:
![image](https://user-images.githubusercontent.com/983050/191746214-30647a8b-9df7-44bd-9e1d-9f8bd773b37d.png)

Not sure we need to handle it 
Any idea @nataliadelmar or @torgeadelin ?

jor-row replied:
To be honest I haven't been testing these styling changes in Chrome as the option to use autofill wasn't coming up for me. I see how to use it now. 

For some reason the animation won't work in chrome, and from trying a few things quickly can't seem to override the styling on chrome to make it transparent like I have in the other two browsers. 

The awkward thing is I can "change" the colour by overlaying a coloured box-shadow but that of course changes it in Safari (which causes the autofill icon colours to not be visible and the text to be less visible) and Firefox. 

In the meantime while addressing the other comments I have styled it so the edges are at least curved and it fits in the text box. 
 
<img width="397" alt="Screenshot 2022-09-22 at 17 55 22" src="https://user-images.githubusercontent.com/86778628/191806999-baf9865d-a540-4b8f-a36e-b8cb7ff3a4f0.png">


Any ideas?

nataliadelmar replied:
Is that a purple? This color doesn't look off to me like the bright yellow did so I am happy to keep it this way, including the rounded corners. It still passes color contrast and is in tone with the rebrand colors, which are only dark mode because this logged-out part of the app is always in dark mode right?

jor-row replied:
Yes, I've been told that these log on pages are always in the default dark mode as light mode is set as a user preference whilst logged in. I can't think of any time within the app that a form would be used with these kinds of entries that would prompt the browser to use autofill, so I think it should be fine? There may, however, be features within the app that I'm unaware of that would do this. 

As for the X in the input field, I hadn't noticed but yes there should be an X in the black square and there will be when it's used in Cantina. The reason there isn't one in my stories is I haven't used the "value" prop of TextInput which tells the component whether to show it or not. To make it work in my stories I'll have to add separate state to each form field to handle it which I'm working on now as I suppose it's worth including.

nataliadelmar replied:
I understand. In my opinion, the short case that a user will not see an X button until clicking on the field in Chrome if there is autofilled data in Chrome only is small enough to let this case go. What do you think @maxinteger ?

maxinteger replied:
I agree with @nataliadelmar this is kind of an edge.

jor-row commented on src/components/TextInput/TextInput.stories.tsx:
```diff
@@ -85,10 +92,64 @@ const BetterExample: FC<StoryProps> = (props: StoryProps) => {
+            style={{ marginBottom: '1rem', width: '16.25rem' }}
+          />
+        )}
+        <ButtonPill size={32} grown style={{ marginBottom: '1rem', width: '16.25rem' }}>
```
I've changed to the text to "Submit". My only reasoning for using "Next" was to mimic the log on pages seen on web.webex.

jor-row commented on src/components/TextInput/TextInput.stories.tsx:
```diff
@@ -85,10 +92,64 @@ const BetterExample: FC<StoryProps> = (props: StoryProps) => {
+  const { displayName = false, emailAddress = false, password = false } = props;
+
+  return (
+    <>
```
That explains a lot, thanks!
